inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -0,0 +1,102 @@
+"""
+    Copyright (C) 2023 Intel Corporation
+    SPDX-License-Identifier: Apache-2.0
+"""
+
+import logging
+import json
+from dispatcher.common.result_constants import Result
+from dispatcher.source.constants import ApplicationRemoveSourceParameters, OsType, SourceParameters
+from dispatcher.source.source_manager_factory import create_os_source_manager
+from dispatcher.source.source_manager_factory import create_application_source_manager
+from inbm_lib.xmlhandler import XmlException, XmlHandler
+
+logger = logging.getLogger(__name__)
+
+
+def do_source_command(parsed_head: XmlHandler, os_type: OsType) -> Result:
+    """
+    Run a source command.
+
+    @param parsed_head: XmlHandler corresponding to the manifest tag
+    @param os_type: os type
+    @return Result
+    """
+    logger.debug(f"do_source_command: {parsed_head}")
+
+    try:
+        os_action = parsed_head.get_children('osSource')
+        if os_action:
+            return _handle_os_source_command(parsed_head, os_type, os_action)
+    except XmlException:
+        pass  # If we get an XmlException here we still want to try applicationSource
```

Comment:
I don't like this.  But I don't have a solution ATM.  We can come back and see if we can find a better OO way.